src/code_actions.zig

```diff
@@ -233,6 +262,37 @@ fn handlePointlessDiscard(builder: *Builder, actions: *std.ArrayListUnmanaged(ty
     });
 }
 
+// attempts to converts a slice of text into camelcase 'FUNCTION_NAME' -> 'functionName'
+fn createCamelcaseText(allocator: std.mem.Allocator, identifier: []const u8) ![]const u8 {
+    var num_separators: usize = 0;
+
+    for (identifier) |c| {
+        if (c == '_') num_separators += 1;
+    }
+
+    const new_text_len = identifier.len - num_separators;
+    var new_text = try std.ArrayListUnmanaged(u8).initCapacity(allocator, new_text_len);
+    errdefer new_text.deinit(allocator);
+
+    var idx: usize = 0;
+
+    // sloppy
+    while (idx < identifier.len) {
+        const c = identifier[idx];
+
+        if (c == '_') {
+            const c2 = identifier[idx + 1];
+            new_text.appendAssumeCapacity(std.ascii.toUpper(c2));
+            idx += 2;
+        } else {
+            new_text.appendAssumeCapacity(std.ascii.toLower(c));
+            idx += 1;
+        }
+    }
```

Comment:
this causes a index out of bound error when you have an identifier which ends with an underscore like
```zig
fn foo_bar_baz_() void {}
```